crates/feldera-types/src/transport/kafka.rs

```diff
@@ -146,6 +149,14 @@ impl KafkaInputConfig {
             .or_insert_with(|| val.to_string());
     }
 
+    fn get_pem_options(&self) -> Vec<(String, String)> {
```

Comment:
I agree with @snkas , this should be in adapters and applied after secret resolution; you should be able to dedup all this logic by putting it in `transport/kafka/mod.rs`.